src/physics.F90

```diff
@@ -828,27 +828,28 @@ subroutine sample_target_velocity(nuc, v_target, E, uvw, v_neut, wgt, xs_eff, kT
     integer :: i_E_low ! 0K index to lowest practical relative energy
     integer :: i_E_up  ! 0K index to highest practical relative energy
     integer :: i_E_rel ! index to trial relative energy
+    integer :: n_grid  ! number of energies on 0K grid
 
     logical :: reject  ! resample if true
 
-    character(80) :: sampling_scheme ! method of target velocity sampling
+    integer :: sampling_method ! method of target velocity sampling
 
     awr = nuc % awr
 
     ! check if nuclide is a resonant scatterer
     if (nuc % resonant) then
 
-      ! sampling scheme to use
-      sampling_scheme = nuc % scheme
+      ! sampling method to use
+      sampling_method = res_scat_method
 
       ! upper resonance scattering energy bound (target is at rest above this E)
-      if (E > nuc % E_max) then
+      if (E > res_scat_energy_max) then
```

Comment:
Here, or somewhere earlier on, we should also make sure that resonance scattering is not done in the URR by either checking that E < E_urr_min (here) or res_scat_energy_max < E_urr_min (earlier on)